config/initializers/middlewares.rb

```diff
@@ -1,3 +1,6 @@
+require_relative "../../app/lib/middlewares/set_cookie_domain"
+require_relative "../../app/lib/middlewares/set_time_zone"
```

Comment:
Looks like [Rails 7 disabled autoload during app initialization](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#autoloading-during-initialization) (the part of the lifecycle where these files run). The solution seems to be to [wrap code that references those constants inside a `Rails.application.config.to_prepare` block](https://guides.rubyonrails.org/v7.0/autoloading_and_reloading_constants.html#use-case-1-during-boot-load-reloadable-code).

Reply:
This is based on my own understanding of `to_prepare`. I could be wrong here.
- On production. If you aren't setting any global variable that might be gone after a reload, you don't need it. Such just regular PORO.
- In development. If you are changing the code (liquid tags, for example) when the server is up, the new code won't be loaded and you'll have to restart the server.

I didn't bother changing this portion that Michael added because it seems to be working fine but having `to_prepare` seems to cause no harm too. I can add it "just in case".

Reply:
I just tried it implement this and I got a `NameError`. My guess is that it doesn't actually execute the code in the block yet.

Reply:
Hmmm, maybe that's why he didn't go that route, either. Thank you for checking!